### PR TITLE
Changes for Python 3.6+ (deprecating Python 2.x)

### PR DESCRIPTION
**Pull request by markpeek**
Switching awacs to Python 3 only by deprecating Python 2.x support. This removes the need for 2to3.

@michael-k could you review and approve?

### CODE REVIEW

File: tests/test_examples.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def add_tests():
     # Filter out all *.py files from the examples directory
     examples = 'examples'
     regex = re.compile(r'.py$', re.I)
-    example_filesnames = filter(regex.search, os.listdir(examples))
+    example_filesnames = list(filter(regex.search, os.listdir(examples)))
```
Comment (michael-k) [inline tests/test_examples.py:30]: `list()` isn't necessary here

File: awacs/__init__.py [hunk 2]
```diff
@@ -18,7 +18,7 @@ def __init__(self, name, type=None, dictname=None, props={}, **kwargs):
         self.name = name
         self.props = props
         # Cache the keys for validity checks
-        self.propnames = props.keys()
+        self.propnames = list(props.keys())
```
Comment (michael-k) [inline awacs/__init__.py:21]: Is `list()` necessary here?  The `in` check in line 50 doesn't need it.
https://docs.python.org/3/library/stdtypes.html#dict-views

Comment (michael-k) [reply awacs/__init__.py:21]: `AWSObject` and `AWSHelperFn` can also stop inheriting from `object` explicitly.

**Review by michael-k — APPROVED** [review]
I love it :)

There's a similar PR for troposphere: cloudtools/troposphere#1812.  It has a lot of conflicts by now, but I'm happy to resolve them if you want to forward with it. :)

**Comment by michael-k** [comment]
If you plan to bump the major version (due to dropping Python 2), I suggest to include the following changes:

- Switch from `setup.py` to `setup.cfg` (declarative).
- Providing wheels via PyPI.
- Switch from pycodestyle to black (esp. autoformatting for generated files).
- Change the default branch from `master` to `main`.

These changes aren't breaking and don't require a major version bump.  But they would be a nice addition.  I'm happy to open PRs for the changes.

File: tests/test_examples.py (hunk 1 above)
Comment (markpeek) [reply tests/test_examples.py]: fixed

File: awacs/__init__.py (hunk 2 above)
Comment (markpeek) [reply awacs/__init__.py]: fixed

**Comment by markpeek** [comment]
> I love it :)
> 
> There's a similar PR for troposphere: [cloudtools/troposphere#1812](https://github.com/cloudtools/troposphere/pull/1812). It has a lot of conflicts by now, but I'm happy to resolve them if you want to forward with it. :)

Thank you @michael-k for getting to the review so quickly. I just pushed changes per your feedback.
Yes, I'm intending to switch troposphere as well to Python3 and I'll comment over on your issue with details.

**Comment by markpeek** [comment]
> If you plan to bump the major version (due to dropping Python 2), I suggest to include the following changes:
Yes, with the Python3 change I will be bumping to a new major version.

> * Switch from `setup.py` to `setup.cfg` (declarative).
> * Providing wheels via PyPI.
I haven't touched the setup and distribution side of things in awhile. Please propose changes for these first two items you listed.

> * Switch from pycodestyle to black (esp. autoformatting for generated files).
This is likely easier for me to handle via a direct commit. Let me look at it.

> * Change the default branch from `master` to `main`.
Good suggestion. I'll look to do this after all the changes are in and before the release is done.

**Comment by michael-k** [comment]
> I haven't touched the setup and distribution side of things in awhile. Please propose changes for these first two items you listed.

I'll send a PR later today. :)

**Pull request merged**